float32 desired
float32[] position
float32[] velocity
float32[] cmd
float32[] effort